### PR TITLE
CHECKOUT-4726 Allow creating customer address

### PR DESCRIPTION
**Pull request by capsula4**
## What?
Allow creating customer address. Alpha functionality.

## Why?
To allow creating customer addresses during checkout, which can be used for the multishipping UI.

## Testing / Proof
unit

@bigcommerce/checkout

### CODE REVIEW

File: src/customer/customer-action-creator.ts [hunk 1]
```diff
@@ -54,6 +54,24 @@ export default class CustomerActionCreator {
...
+    createAddress(
+        customerAddress: CustomerAddressRequestBody,
+        options?: RequestOptions
+    ): ThunkAction<CreateCustomerAddressAction, InternalCheckoutSelectors> {
+        return store => concat(
+            of(createAction(CustomerActionType.CreateCustomerAddressRequested)),
+            from(this._customerRequestSender.createAddress(customerAddress, options))
+                .pipe(
+                    switchMap(() => concat(
+                        this._checkoutActionCreator.loadCurrentCheckout(options)(store),
```
Comment (davidchin) [inline src/customer/customer-action-creator.ts:66]: I wonder if the customer address endpoint itself can return all the newly created address, we can then map the response in the same shape as the checkout's customer sub resource and store it in memory. That way we don't have to make an extra call to hydrate the cart, which can sometimes be slow depending on its size.

**Review by davidchin — APPROVED** [review]

File: src/customer/customer-action-creator.ts (hunk 1 above)
Comment (capsula4) [reply src/customer/customer-action-creator.ts:66]: Yeah, that would be great, or have a customer endpoint where we can retrieve addresses, avoiding having to reload the cart/checkout.

Comment (davidchin) [reply src/customer/customer-action-creator.ts:66]: cc @snaderiBC

Comment (snaderiBC) [reply src/customer/customer-action-creator.ts:66]: >That way we don't have to make an extra call to hydrate the cart

That's interesting idea 🤔

**Pull request merged**